framework/src/mesh/FileMesh.C

```diff
@@ -118,6 +119,18 @@ FileMesh::buildMesh()
         mooseError("cannot locate mesh file '", _file_name, "'");
       getMesh().read(_file_name);
 
+      // we also read mesh meta data here if there is any
+      RestartableDataIO restartable(_app);
+      restartable.setErrorOnLoadWithDifferentNumberOfProcessors(false);
+      std::string fname = _file_name + "/mesh_meta_data" + restartable.getRestartableDataExt();
+      if (MooseUtils::pathExists(fname))
+      {
+        // get reference to mesh meta data (created by MooseApp)
+        auto & meta_data = _app.setRestartableDataMap(MooseApp::MESH_META_DATA);
```

Comment:
This is where `setRestartableDataMap` is used. If this code is in mesh generators instead of `FileMesh`, we can load the mesh meta data locally and then let the mesh generator to push them into app with `declareMeshProperty`.

Reply:
Avoiding the new mutator would be great

Reply:
`readRestartableData` takes a `const RestartableDataMaps &` not a `RestartableDataMaps &`, so AFAICT there is no need for the new mutator. You should be able to just use the existing `getRestartableDataMap`

Reply:
You are right, but how is this possible? Supposed we are loading the data into `meta_data`.

Reply:
Oh they are loaded into RestartableDataValuePair which is using `std::unique_ptr` to hold the data.